src/classes/CW_AudienceGroupTriggerHandlerTest.cls

```diff
@@ -0,0 +1,236 @@
+/**
+* File Name   :    CW_AudienceGroupTriggerHandlerTest.cls
+* Description :    D-106803o
+*                 
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 24/04/2014 Alex  Panka		Initial
+ */
+ 
+@isTest
+public with sharing class CW_AudienceGroupTriggerHandlerTest {
+	
+	static testMethod void ValidationRuleTest() {
+		
+		CW_Campaign__c camp = TestDataUtil.createCampaign();
+      insert camp;
+      //We muset use single call for testDataUtil for uniq names
+      List<CW_Audience_Group__c> listAllGrps  =TestDataUtil.createAudienceGroups(camp.Id,6);
+      
+		List<CW_Audience_Group__c> listControlGrps  = new List <CW_Audience_Group__c>();
+		List<CW_Audience_Group__c> listControlGrps2 = new List <CW_Audience_Group__c>();
+		List<CW_Audience_Group__c> listNormalGrps   = new List <CW_Audience_Group__c>();
+		listControlGrps.Add(listAllGrps.get(0));
+		listControlGrps.Add(listAllGrps.get(1));
+		listControlGrps2.Add(listAllGrps.get(2));
+		listControlGrps2.Add(listAllGrps.get(3));
+		listNormalGrps.Add(listAllGrps.get(4));
+		listNormalGrps.Add(listAllGrps.get(5));
+		
+		insert listControlGrps;
+		insert listControlGrps2;
+		
+
+
+		for(integer i=0;i<2;i++){
+			listNormalGrps.get(i).Is_Control_Group__c=false;
+			listNormalGrps.get(i).Control_Group__c=listControlGrps.get(i).id;
+		}
+		insert listNormalGrps;
+	   //Test if we can chage witout ref
+	   for(CW_Audience_Group__c Rec :listControlGrps2){
+	   	Rec.Is_Control_Group__c=false;
+	   }
+	   update listControlGrps2;
+	   for(CW_Audience_Group__c Rec :listControlGrps2){
+	   	Rec.Is_Control_Group__c=true;
+	   }
+		update listControlGrps2;
+	   
+	   try{
+	   	insert listAllGrps;
+	   	System.assert(false,'Validation of Unique Name on Audience control group not rise error');
+	   }
+	   catch(Exception E){}	 
+	   
+	   
+	}
+	
+	static testMethod void controlGroupTest()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility